lua/entities/base_wire_entity.lua

```diff
@@ -324,28 +332,67 @@ if CLIENT then return end -- no more client
 --------------------------------------------------------------------------------
 
 util.AddNetworkString( "wire_overlay_data" )
+util.AddNetworkString( "wire_overlay_request" )
 
-timer.Create("WireOverlayUpdate", 0.1, 0, function()
-	for _, ply in ipairs(player.GetAll()) do
-		local ent = ply:GetEyeTrace().Entity
-		if IsValid(ent) and ent.IsWire and
-			ent.OverlayData and
-			ent.OverlayData_UpdateTime and
-			ent.OverlayData_UpdateTime.time > (ent.OverlayData_UpdateTime[ply] or 0) then
+--------------------------------------------------------------------------------
+-- Other functions
+--------------------------------------------------------------------------------
 
-			ent.OverlayData_UpdateTime[ply] = CurTime()
+-- this table keeps a list of players looking at each entity
+-- table structure: overlayRequests[ent] = {ply1,ply2,...}
+local overlayRequests = {}
 
-			net.Start( "wire_overlay_data" )
-				net.WriteEntity( ent )
-				net.WriteTable( ent.OverlayData )
-			net.Send( ply )
+local function syncWireOverlay( ent, use_udp, players )
+	if not players then 
+		players = overlayRequests[ent]
```

Comment:
players needs to be checked for null players. If one disconnects will looking at a wire entity, expect lua errors. This is why I liked my implementation because I just looped over active players.

Reply:
yes correct I forgot about that in my changes. I'll add that check back in tomorrow unless you want to do it before then

Reply:
This would also be a good place to use tomy's PR <https://github.com/wiremod/wire/pull/2110>

Reply:
Perhaps, but that is indexed by player rather than by number.

Reply:
It could, however, be used in my version so that I'm not looping over every player.